slicedimage/io.py

```diff
@@ -34,10 +34,29 @@ def infer_backend(baseurl, allow_caching=True):
     return backend
 
 
-def resolve_url(name_or_url, baseurl=None, allow_caching=True):
+def resolve_path_or_url(path_or_url, allow_caching=True):
+    """
+    Given either a path (absolute or relative), or a URL, attempt to resolve it.  Returns a tuple consisting of:
+    a :py:class:`slicedimage.backends._base.Backend`, the basename of the object, and the baseurl of the object.
+    """
+    try:
+        return resolve_name_or_url(path_or_url, allow_caching=allow_caching)
+    except ValueError:
+        if os.path.isfile(path_or_url):
+            return resolve_name_or_url(
+                os.path.basename(path_or_url),
+                baseurl="file://{}".format(os.path.dirname(os.path.abspath(path_or_url))),
+                allow_caching=allow_caching,
+            )
+        raise
+
+
+def resolve_name_or_url(name_or_url, baseurl=None, allow_caching=True):
```

Comment:
they are semantically different, though there's a lot of overlap.

`resolve_name_or_url` is more along the line of: either a URL or a baseurl+name.  this is along the line of how a browser resolves a url on a webpage.  it can either be a relative URL (base url + name) or an absolute URL.

`resolve_path_or_url` is **only** absolute URL resolution where we special case non-URLs as filesystem URLs.  this is mostly for simplifying how human beings enter locations.

Reply:
I think there's a lot of potential improvements in the name of the method or the documentation, but I don't see an obvious way to combine the two.

For instance, if starfish used `resolve_path_or_url` instead of `resolve_name_or_url` in https://github.com/spacetx/starfish/pull/146, then one could input a filesystem path for hybridization.json and it would be accepted.

Reply:
`resolve_absolute_path_or_url` and `resolve_relative_path_or_url`?

Reply:
Well... that's not exactly it either.